### PR TITLE
Fix regressions from #6610

### LINKED ISSUE #7031: Program denies validity of most audio formats in internal browser popup
### Bug Summary
When using the load sample prompt on basically anything, only available audio files are .ds and .aiff. Surprisingly the samples added from sidebar still are fine and even load right.
#### Steps to reproduce
Place audio-using internal plugin (AudioFileProcessor, SlicerT, Ect), click sample load button, grieve,
#### Expected behavior
All usual available sample formats show (and hopefully eventually mp3 too but who knows when that'll work right)
#### Actual behavior
Only shows .ds and .aiff. what even is .ds, who uses that.
#### Screenshot
image

#### Affected LMMS versions
1.3.0-alpha.1.489+g4e63f6065

### LINKED ISSUE #7032: Some samples crashing program
### Bug Summary
When trying to select some specific samples (Example in my case is a bunch of FL Studio's factory samples (Not all though apparently)), the program will freeze a moment then crash
#### Steps to reproduce
This probably applies to more, but I just have FL's for reference, so have those, try selecting some (Some seem to work others don't, I know some SFX and Drum stuff doesn't work), and then watch your program crash
#### Expected behavior
The sound selects properly, and lets me use it
#### Actual behavior
Say bye to your current instance
#### Affected LMMS versions
1.3.0-alpha.1.489+g4e63f6065

### PR DESCRIPTION
**Pull request by sakertooth**
This PR fixes two regressions from #6610, that being:
1. Certain samples crash LMMS when calling `decodeSampleOggVorbis`
2. Only AIFF and DrumSynth are listed as file formats

Fixes #7031
Fixes #7032

### CODE REVIEW

File: src/core/SampleDecoder.cpp [hunk 1]
```diff
@@ -116,15 +116,46 @@ auto decodeSampleDS(const QString& audioFile) -> std::optional<SampleDecoder::Re
 #ifdef LMMS_HAVE_OGGVORBIS
 auto decodeSampleOggVorbis(const QString& audioFile) -> std::optional<SampleDecoder::Result>
 {
-	auto vorbisFile = OggVorbis_File{};
-	const auto openError = ov_fopen(audioFile.toLocal8Bit(), &vorbisFile);
+	static auto s_read = [](void* buffer, size_t size, size_t count, void* stream) -> size_t {
+		auto file = static_cast<QFile*>(stream);
```
Comment (Veratil) [inline src/core/SampleDecoder.cpp:120]: Maybe out of scope, but since this is core should we try to not use `QFile` here?

Comment (sakertooth) [reply src/core/SampleDecoder.cpp:120]: `QFile` is used to seamlessly handle incompatible file paths between Windows and other POSIX systems. Windows uses UTF-16 for encoding their file paths, while most APIs and POSIX use UTF-8, so if we hadn't used something like `QFile`, we would need to manually convert the file path to a valid UTF-16 encoding for Windows then call some wide string variant of `fopen`, which I don't even think Vorbisfile has.

Instead, it is much easier if we abstract away from the handling of file paths on different systems and just work through `QFile` instead by opening the file there and then using that.

Comment (Veratil) [reply src/core/SampleDecoder.cpp:120]: So pain later when removing it from core. :)

Comment (sakertooth) [reply src/core/SampleDecoder.cpp:120]: > So pain later when removing it from core. :)

Do you know of any cleaner alternative?

Comment (Veratil) [reply src/core/SampleDecoder.cpp:120]: Clean isn't the issue here, it's dependency management. We can't leave this in core in the long term.

Comment (sakertooth) [reply src/core/SampleDecoder.cpp:120]: > Clean isn't the issue here, it's dependency management. We can't leave this in core in the long term.

Yes, but `QFile` is the most straightforward solution as of right now. The plan is to move to `std::filesystem` once our CI fully supports it, so we can use that then, and is what I would suggest. At that point, we wouldn't be so dependent on `QFile` and can provide our own compatability layer for file paths on Windows with other systems.

But we don't have `std::fillesystem` in its entirety yet, only in its experimental form, and I don't want to raise complexity by adding experimental and incomplete support for something we will remove anways in place of the full support, which we will do very soon anyways.

Comment (Veratil) [reply src/core/SampleDecoder.cpp:120]: That's why I asked with `try` specifically. As long as we have a path forward in the future to replace it that's okay.

Comment (sakertooth) [reply src/core/SampleDecoder.cpp:120]: I will add a TODO comment though so that we can remember to come back and change it.

**Pull request merged**